osu.Server.Spectator/Hubs/MultiplayerHub.cs

```diff
@@ -170,6 +172,8 @@ private async Task<MultiplayerRoom> retrieveRoom(long roomId)
                 var playlistItem = await db.GetCurrentPlaylistItemAsync(roomId);
                 var beatmapChecksum = await db.GetBeatmapChecksumAsync(playlistItem.beatmap_id);
 
+                Debug.Assert(beatmapChecksum != null);
```

Comment:
Are you sure about using an `Assert` here rather than throwing an exception? It's going to allow deployed builds to get in a potentially bad state (as it will be ignored). See below where it's throwing an exception in the same case.

Note that it should be considered a server-side exception at this point (so not an `InvalidStateException`, since the user doesn't need to know what went wrong).